modules/text/src/ocr_holistic.cpp

```diff
@@ -0,0 +1,1064 @@
+#include "precomp.hpp"
+#include "opencv2/imgproc.hpp"
+#include  "opencv2/highgui.hpp"
+#include "opencv2/core.hpp"
+
+
+
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <queue>
+#include <algorithm>
+#include <iosfwd>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+
+#ifdef HAVE_CAFFE
+#include "caffe/caffe.hpp"
+#endif
+
+#ifdef HAVE_DNN
+#include "opencv2/dnn.hpp"
+#endif
+
+using namespace cv;
+using namespace cv::dnn;
+using namespace std;
+namespace cv { namespace text {
+
+//Maybe OpenCV has a routine better suited
+inline bool fileExists (String filename) {
+    std::ifstream f(filename.c_str());
+    return f.good();
+}
+
+//************************************************************************************
+//******************   ImagePreprocessor   *******************************************
+//************************************************************************************
+
+void ImagePreprocessor::preprocess(InputArray input,OutputArray output,Size sz,int outputChannels){
+    Mat inpImg=input.getMat();
+    Mat outImg;
+    this->preprocess_(inpImg,outImg,sz,outputChannels);
+    outImg.copyTo(output);
+}
+void ImagePreprocessor::set_mean(Mat mean){
+
+
+    this->set_mean_(mean);
+
+}
+
+
+
+class ResizerPreprocessor: public ImagePreprocessor{
```

Comment:
I would separate out the preprocessing code out of this file. It distracts away from the real core of the network setup. Also, the different `preprocess_` functions requires refactoring. They can be achieved in smaller lines of code. Right now, they are too verbose and repetitive.